paasta_tools/list_marathon_service_instances.py

```diff
@@ -25,34 +25,114 @@
 
 - -d <SOA_DIR>, --soa-dir <SOA_DIR>: Specify a SOA config dir to read from
 - -c <CLUSTER>, --cluster <CLUSTER>: Specify which cluster of services to read
+- -m, --minimal: Only show service instances that need bouncing
 """
 import argparse
 import sys
 
 from paasta_tools.marathon_tools import DEFAULT_SOA_DIR
+from paasta_tools.marathon_tools import get_marathon_client
+from paasta_tools.marathon_tools import get_num_at_risk_tasks
+from paasta_tools.marathon_tools import load_marathon_config
+from paasta_tools.marathon_tools import load_marathon_service_config
 from paasta_tools.utils import compose_job_id
+from paasta_tools.utils import decompose_job_id
 from paasta_tools.utils import get_services_for_cluster
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import NoDockerImageError
+from paasta_tools.utils import use_requests_cache
 
 
 def parse_args():
-    parser = argparse.ArgumentParser(description='Lists marathon instances for a service.')
+    parser = argparse.ArgumentParser(
+        description='Lists marathon instances for a service.')
     parser.add_argument('-c', '--cluster', dest="cluster", metavar="CLUSTER",
                         default=None,
                         help="define a specific cluster to read from")
     parser.add_argument('-d', '--soa-dir', dest="soa_dir", metavar="SOA_DIR",
                         default=DEFAULT_SOA_DIR,
                         help="define a different soa config directory")
+    parser.add_argument('-m', '--minimal', dest='minimal', action='store_true',
+                        help="show only service instances that need bouncing")
     args = parser.parse_args()
     return args
 
 
+def long_job_id_to_short_job_id(long_job_id):
```

Comment:
This is "sorta" duplicating `get_matching_apps()`, which takes a service/instance and outputs the apps that match. (as opposed to looking at all the apps and reversing them)

Once nice approach to using `get_matching_apps` is that you only "look at" the ones that are legit.

How will this code tolerate the condition where a service was just deleted from soa-configs?